### DISCUSSION TITLE
Please help me fix my code

### DISCUSSION THREAD

**Discussion by ghost** [koreader/koreader#11947] [Q&A]
https://github.com/koreader/koreader/issues/11935#issue-2328081543

In one specific book, the indents are huge, I wrote the code, but apparently it is incorrect, because after it the indents became smaller, but still large indents compared to other books.

```
/* User style tweak для KOReader */
.calibre p, .calibre div, .calibre2 p, .calibre2 div, .Style, .calibre16 {
  text-indent: 0em !important; /* Убирает отступ первой строки абзаца */
  margin-top: 0em !important; /* Убирает верхний отступ абзаца */
  margin-bottom: 0em !important; /* Убирает нижний отступ абзаца */
}

```

![IMG_20240531_200919.jpg](https://github.com/koreader/koreader/assets/171361008/1eb729d4-444d-41eb-8993-0233ab0f75a2)

![IMG_20240531_200928.jpg](https://github.com/koreader/koreader/assets/171361008/b8e495ce-0437-439a-8641-a9de9b52909d)

**Comment by poire-z** [comment]
Duplicate of https://github.com/koreader/koreader/issues/11935

**Reply by ghost** [reply]
> Duplicate of https://github.com/koreader/koreader/issues/11935

Can you PLEASE help me again?
There is a new problem with that, similar:
Suddenly there is too much space (or indentation) in the middle of the text. What code should I add here (in addition to the first)? Look at photos:

![IMG_20240601_135606.jpg](https://github.com/koreader/koreader/assets/171361008/54b9bd99-25d7-4866-89b9-1f007ae8c81b)

![IMG_20240601_135613.jpg](https://github.com/koreader/koreader/assets/171361008/1723dce7-dc09-4ef9-9fef-0055b383bfed)

Please please help me 🙏 😭

**Reply by poire-z** [reply]
You can't solve this case in KOReader: the consecutive non-breaking-spaces are among content text, there is not tag wrappping it that we would target.
Your only solution is to fix the crappy book elsewhere on your computer. May be calibre can clean up such books.
